### ISSUE TITLE
Custom :focus styles

### ISSUE THREAD

**Issue by asangma** [Esri/calcite-components#20]
Are there any plans or patterns to do custom :focus states?
We sometimes see weird stuff with browser default focus outline.

Of course, we would want to retain accessibility.

I know there's stuff like outliner.js and stuff like that.

cc @driskull @jcfranco 

[a11yproject.com article](https://a11yproject.com/posts/never-remove-css-outlines/)

**Comment by paulcpederson** [comment]
@bstifle do you have a set of designs for focus states library-wide? If we can get away with a single custom focus style everywhere that would be better, but we could also do one-off focus styles for every element as long as they're visually cohesive.

**Comment by macandcheese** [comment]
cc @asangma and Mitch re out conversation yesterday as well

**Comment by vcolavin** [comment]
Just to note, there is an open issue related to this in the Share dialog.

**Comment by vcolavin** [comment]
https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/26747

**Comment by macandcheese** [comment]
@asangma did you all have any thoughts here?

**Comment by bstifle** [comment]
![Focus States Light Simple](https://user-images.githubusercontent.com/14083530/72574628-85cc1280-387e-11ea-9185-2b6b41605d7f.png)
![Examples Light](https://user-images.githubusercontent.com/14083530/72574631-8795d600-387e-11ea-9640-79738e019f0a.png)
![Examples Dark](https://user-images.githubusercontent.com/14083530/72574632-8795d600-387e-11ea-89f4-983299982494.png)


@asangma @julio8a @paulcpederson @macandcheese

Here is the latest update based on a few meetings and talks with folks. See the Rules png pasted here. 

- It is a basic cascading rule set using increments of 2. 
- Light theme/dark theme.
- Simple transitions
- See this link for a super quick demo https://codepen.io/bslayer/pen/qBEMgQw


Next steps will be getting a more high fidelity demo, and then testing. 

After that, Ideally we will test 2 designs. A) Being the styles shown here and B) being an even more super noticeable color way (black and white instead of blue)

**Comment by bstifle** [comment]
There's 3 main things we are trying to solve with this design, weighted by importance:

1. Consistency across Esri properties and components
2. Noticeable, meaningful, and WCAG-able—While not being overly clunky and distracting.
3. Scalability and ease of use for current and future designs

**Comment by driskull** [comment]
Looks good!

**Comment by asangma** [comment]
Nice!
![slicer](https://user-images.githubusercontent.com/12503298/72575578-beb9b680-3881-11ea-80c8-923647c653f0.gif)

**Comment by bstifle** [comment]
moving this convo over at calcite-base https://github.com/Esri/calcite-base/issues/19
